crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -595,3 +596,685 @@ async fn cow_amm_driver_support(web3: Web3) {
+    // the user order.
+
+    // Set up the CoW AMM as before
+    let oracle = contracts::CowAmmUniswapV2PriceOracle::builder(&web3)
+        .deploy()
+        .await
+        .unwrap();
+
+    let cow_amm_factory = contracts::CowAmmConstantProductFactory::builder(
+        &web3,
+        onchain.contracts().gp_settlement.address(),
+    )
+    .deploy()
+    .await
+    .unwrap();
+
+    // Fund the CoW AMM owner with DAI and WETH and approve the factory to transfer
+    // them
+    dai.mint(cow_amm_owner.address(), to_wei(2_000)).await;
+    tx!(
+        cow_amm_owner.account(),
+        dai.approve(cow_amm_factory.address(), to_wei(2_000))
+    );
+
+    tx_value!(
+        cow_amm_owner.account(),
+        to_wei(1),
+        onchain.contracts().weth.deposit()
+    );
+    tx!(
+        cow_amm_owner.account(),
+        onchain
+            .contracts()
+            .weth
+            .approve(cow_amm_factory.address(), to_wei(1))
+    );
+
+    tx_value!(
+        solver.account(),
+        to_wei(1),
+        onchain.contracts().weth.deposit()
+    );
+
+    // Fund the settlement contract with WETH so it can pay out the user order.
+    tx!(
+        solver.account(),
+        onchain
+            .contracts()
+            .weth
+            .transfer(onchain.contracts().gp_settlement.address(), to_wei(1))
+    );
+
+    let pair = onchain
+        .contracts()
+        .uniswap_v2_factory
+        .get_pair(onchain.contracts().weth.address(), dai.address())
+        .call()
+        .await
+        .expect("failed to get Uniswap V2 pair");
+
+    let cow_amm_address = cow_amm_factory
+        .amm_deterministic_address(
+            cow_amm_owner.address(),
+            dai.address(),
+            onchain.contracts().weth.address(),
+        )
+        .call()
+        .await
+        .unwrap();
+
+    let oracle_data: Vec<_> = std::iter::repeat(0u8)
+        .take(12) // pad with 12 zeros to end up with 32 bytes
+        .chain(pair.as_bytes().to_vec())
+        .collect();
+    const APP_DATA: [u8; 32] = [12u8; 32];
+
+    // Create the CoW AMM
+    cow_amm_factory
+        .create(
+            dai.address(),
+            to_wei(2_000),
+            onchain.contracts().weth.address(),
+            to_wei(1),
+            0.into(), // min traded token
+            oracle.address(),
+            ethcontract::Bytes(oracle_data.clone()),
+            ethcontract::Bytes(APP_DATA),
+        )
+        .from(cow_amm_owner.account().clone())
+        .send()
+        .await
+        .unwrap();
+    let cow_amm = contracts::CowAmm::at(&web3, cow_amm_address);
+
+    // Start system with the regular baseline solver as a quoter but a mock solver
+    // for the actual solver competition. This allows us to handcraft a solution.
+    let mock_solver = Mock::default();
+    colocation::start_driver(
+        onchain.contracts(),
+        vec![
+            colocation::start_baseline_solver(
```

Comment:
We are not using the regular baseline solver for anything anymore, right? Then this can be removed and the comment above be amended.

Reply:
It is used for native price fetching.

Reply:
https://github.com/cowprotocol/services/blob/af9d499618e4e01b22952cd5d240e42b93d9442e/crates/e2e/src/setup/services.rs#L107-L109